maxim/apis/maxim_apis.py

```
@@ -110,47 +117,187 @@ def __init__(self, base_url: str, api_key: str):
+                    scribe().warning(
+                        f"[MaximSDK] Connection error on attempt {attempt + 1}/{max_retries + 1}: {e}. Retrying in {delay:.1f}s..."
+                    )
+                    time.sleep(delay)
+                    continue
+                else:
+                    scribe().error(
+                        f"[MaximSDK] Failed to establish connection after {max_retries + 1} attempts. Last error: {e}"
+                    )
+                    raise Exception(
+                        f"Connection failed after {max_retries + 1} attempts: {e}"
+                    )
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)